File: framework/math/bounds/Sphere.java
package framework.math.bounds;

import framework.math.Vector3;

public class Sphere extends Bounds {

    public final Vector3 center;
    public float radius;


    public Sphere(float x, float y, float z, float radius) {
        super();
        this.center = new Vector3(x, y, z);
        this.radius = radius;
    }
}

File: C++/GameEngineCpp/src/physics/Sphere.cpp
#include "include/physics/3d-colliders/Sphere.hpp"

#include <math/MathUtils.hpp>

using namespace GameEngine::Physics::Bounds;


Sphere::Sphere(const Vec3& center, double radius)
    : center(center), radius(radius) {}

bool Sphere::operator!=(const Sphere& other) const
{
    return center != other.center || !isZero(radius - other.radius);
}

bool Sphere::operator==(const Sphere& other) const
{
    return center == other.center && isZero(radius - other.radius);
}

bool Sphere::contains(const Vec3& point) const
{
    return center.distSq(point) <= radius * radius;
}
